addons/html_builder/static/src/builder/plugins/visibility_plugin.js

```diff
@@ -0,0 +1,64 @@
+import { Plugin } from "@html_editor/plugin";
+import { isMobileView } from "@html_builder/builder/utils/utils";
+
+export class VisibilityPlugin extends Plugin {
+    static id = "visibility";
+    resources = {
+        on_target_show: this.onTargetShow.bind(this),
+        on_target_hide: this.onTargetHide.bind(this),
+    };
+
+    onTargetHide(snippetEl) {
+        switch (getSelectorName(snippetEl)) {
+            case "DeviceVisibility": {
+                hideInvisibleEl(snippetEl);
+                return;
+            }
+            case "ConditionalVisibility": {
+                hideInvisibleEl(snippetEl);
+                if (snippetEl.classList.contains("o_snippet_invisible")) {
+                    snippetEl.classList.add("o_conditional_hidden");
+                }
+                return;
+            }
+        }
+    }
+    onTargetShow(snippetEl) {
+        switch (getSelectorName(snippetEl)) {
+            case "DeviceVisibility": {
+                showInvisibleEl(snippetEl);
+                return;
+            }
+            case "ConditionalVisibility": {
+                showInvisibleEl(snippetEl);
+                snippetEl.classList.remove("o_conditional_hidden");
+                return;
+            }
+        }
+    }
+}
+
+function getSelectorName(snippetEl) {
+    if (snippetEl.matches("section .row > div")) {
+        return "DeviceVisibility";
+    }
+    if (snippetEl.matches("section, .s_hr")) {
+        return "ConditionalVisibility";
+    }
+}
```

Comment:
I think we should not duplicate the selectors across files, it is error prone (we could forget to change one when modifying the options later for example)